pkg/controller/template/render.go

```diff
@@ -606,3 +607,68 @@ func urlPort(u string) (interface{}, error) {
 		return "", fmt.Errorf("unknown scheme in %s", u)
 	}
 }
+
+func isOpenShiftManagedDefaultLBType(lbType configv1.PlatformLoadBalancerType) bool {
+	switch lbType {
+	case configv1.LoadBalancerTypeOpenShiftManagedDefault:
+		return true
+	default:
+		return false
+	}
+}
+
+func isOpenShiftManagedDefaultLB(cfg RenderConfig) bool {
+	if cfg.Infra.Status.PlatformStatus != nil {
+		lbType := configv1.LoadBalancerTypeOpenShiftManagedDefault
+		switch cfg.Infra.Status.PlatformStatus.Type {
+		case configv1.BareMetalPlatformType:
+			if cfg.Infra.Status.PlatformStatus.BareMetal != nil {
+				if cfg.Infra.Status.PlatformStatus.BareMetal.LoadBalancer != nil {
+					lbType = cfg.Infra.Status.PlatformStatus.BareMetal.LoadBalancer.Type
+				}
+				return isOpenShiftManagedDefaultLBType(lbType)
+			}
+		case configv1.OvirtPlatformType:
+			if cfg.Infra.Status.PlatformStatus.Ovirt != nil {
+				if cfg.Infra.Status.PlatformStatus.Ovirt.LoadBalancer != nil {
+					lbType = cfg.Infra.Status.PlatformStatus.Ovirt.LoadBalancer.Type
+				}
+				return isOpenShiftManagedDefaultLBType(lbType)
+			}
+		case configv1.OpenStackPlatformType:
+			if cfg.Infra.Status.PlatformStatus.OpenStack != nil {
+				if cfg.Infra.Status.PlatformStatus.OpenStack.LoadBalancer != nil {
+					lbType = cfg.Infra.Status.PlatformStatus.OpenStack.LoadBalancer.Type
+				}
+				return isOpenShiftManagedDefaultLBType(lbType)
+			}
+		case configv1.VSpherePlatformType:
+			if cfg.Infra.Status.PlatformStatus.VSphere != nil {
+				emptyAPIInternalIPs := len(cfg.Infra.Status.PlatformStatus.VSphere.APIServerInternalIPs) == 0
+				lbSetNoDefault := (cfg.Infra.Status.PlatformStatus.VSphere.LoadBalancer != nil && cfg.Infra.Status.PlatformStatus.VSphere.LoadBalancer.Type != configv1.LoadBalancerTypeOpenShiftManagedDefault)
+
+				// vSphere allows to use a user managed load balancer by not setting the VIPs in PlatformStatus.
+				// To maintain backward compatibility, we'll follow this logic:
+				// - If the VIPs are not set in PlatformStatus, we assume that the load balancer is managed by a user managed load balancer (existing and well-known vsphere UPI behavior).
+				// - If the VIPs are set in PlatformStatus, we will check if the LoadBalancer Type is set.
+				//   - If the LoadBalancer Type is not set, we will deploy the OpenShift managed load balancer to maintain backward compatibility.
+				//   - If the LoadBalancer Type is set to another value than the default, we will not deploy the internal load balancer.
+				if emptyAPIInternalIPs || lbSetNoDefault {
```

Comment:
@rvanderp3 @jcpowermac question for you guys:

if API VIPs are empty in `PlatformStatus` && `loadBalancer` is explicitly set to `OpenShiftManaged` (meaning we want HAproxy/Keepalived managed in OpenShift), do we allow this?
Or should we require the VIPs in that case? I think yes, we should make sure the VIPs are set if the user wants the openshift managed LB, but I want to confirm with you.

Reply:
I am kinda thinking that should be caught in the installer, wdyt?
Unless lb can be changed day 2?

Reply:
LB can't be changed on L2 for now but will be in the future.

As for the installer, yes I agree and this is why I'm waiting for your answer so I can fix https://github.com/openshift/installer/pull/6812/files#diff-3f811ba555e4ce01f184d351af28272069d04129a14be43b6ec1823be8b38221R1044-R1047

Based on your wish i'll update both MCO & installer.

Reply:
To rephrase, should we requires VIPs to be set in install-config, when we want to use the OpenShift managed LB? I think we have to, right?

Reply:
Answered in async, we require the VIPs in the scenario of OpenShiftManaged LB.